packages/compass-components/src/components/signal-popover.tsx

```diff
@@ -328,8 +348,17 @@ const SignalPopover: React.FunctionComponent<SignalPopoverProps> = ({
 
   return (
     <InteractivePopover
-      className={popoverStyles}
-      containerClassName={popoverContentContainerStyles}
+      className={cx(
+        popoverStyles,
+        // If trigger is not visible, we are in this weird state where trigger
+        // component is hidden by something, but the popover might still be on
+        // the screen. We already started closing popover, but because
+        // leafygreen provides animations that you can't disable, this means
+        // that the popover dissapearance can't be correctly animated in this
+        // case and so we just hide it with styles
+        !triggerVisible && popoverHiddenStyles
```

Comment:
This I am going to steal for Guide Cue :) 

this will avoid the flicker that happens when we leave some components mounted (which had GuideCue displayed) but hidden.


https://github.com/mongodb-js/compass/assets/1305718/7447b837-4f64-4742-9bb4-ddd88ae5e8ef

Reply:
Was about to ping you on that, yeah 😄